### DISCUSSION TITLE
Custom Taxonomies - Filters and Swatches

### DISCUSSION THREAD

**Discussion by justdudeone** [pods-framework/pods#6804] [Ideas]
Hello,

Using of Pods at most cases means that admin need to create some kind of big custom data database - like Cars catalogue, Traveling guide, etc. Most of such databases definitely needs useful search by criterions...

If you use Pods and want to filter Pods Custom Posts by Pods Custom Taxonomies - you can use some 3-rd party filters plugins as FacetWP, Filter Everything, MDTF - Wordpress Meta Data & Taxonomies Filter, etc... but they works good enough only with simple text/numeric Taxonomies.

But if you want to use any kind of visual representation of your Taxonomies as colors, icons, images, etc - you don't have any full-covered option here, because:
A) If you will use any filter which support swatches - you will not see those swatches when Pods display it Relationship Taxonomies fields data.
B) If Pods developers will add swatches support at Pods itself - you will not see those swatches at 3-rd party filters.

So here is only 2 good solutions:
1) Pods developers will add full-covered functionality at Pods itself: add swatches to Taxonomies (images already can be added, it remains to add icons and color); add functions to display swatches at Pods Relationship Taxonomies fields; add display swatches at Filters.
2) More simple - add support for some 3-rd party Swatches+Filters plugins to display their Swatches on Pods Relationship Taxonomies data displaying.

Will be nice to add new Pods Field "Media -> Icon" which will bring possibility to add any icon to each Custom Taxonomy from any font previously uploaded to WP Media or provided with Pods.

Please, provide your decision or ETA, I very need this functionality asap. :\

PS At this aspect "Crokoblock" already make it - it have they Custom Data/Posts/Taxonomies and Filter with swatches also. But it more oriented to work with Elementor, what for me is bad, because Elementor is too bulky.

PPS Most functional samples of swatches+filters - they looks awesome! Pods really needs something like this:
https://plugins.yithemes.com/yith-woocommerce-ajax-product-filter/woman-category/
https://woocommerce-products-filter.berocket.com/shop/
next even with AJAX fast search:
https://demos.xforwoocommerce.com/product-filter/fashion-clothing/
https://woobewoo.com/plugins/woocommerce-filter/

**Comment by justdudeone** [comment]
I have researched and thought about this whole task again and make conclusion that the best way to implement Swatches (Images/Icons/Colors) to Pods with Filtering possibility is to:
+ (I) Add to Pods Media fields section the new "Icon" and "Color" fields.
+ (II) Ask FacetWP developer (still it's the best WP Filter) to add simple improvements:
++ or as already existed field "Color" add new multi-choice fields "Icon" and "Image" with possibility to specify Custom Taxonomy Meta which keep/link correspond data (to not only auto-take it from default Meta=Name as at existed field "Color");
++ or add to fields "Checkbox, Radio, Hierarchy" possibility to choose type image/icon/color (then field "Color" can be removed);
++ or even better FacetWP rework will be to replace all "Checkbox, Radio, Hierarchy" fields with 1 "Uni-Select" field with options: Text select one (Radio), Text select many (Checkboxes), Image select one (1 border), Images select many (Borders with checkboxes), Icon select one (change color of 1), Icons select many (change colors of many), Color select one (1 border), Colors select many (Borders with checkboxes). Hierarchy as second option. Such 1 universal field can cover all those functions! And it will be much simple and fast for users to change filter style (text-to-images, one-to-many, etc, also it can be useful for later dynamic data filtering ;)).

Anyway it seems like the best and most simple and fast way to implement those features - is collaboration of Pods and FacetWP developers!

**Comment by justdudeone** [comment]
Very bad news comes from filtering ans search plugins FacetWP,AJAXSearchPro,etc - NOT ONE of them does NOT support Pods ACT (data stored at custom tables)...
Thus this option without filtering and without search - becomes almost unusable!!!
Again, look at "Crokoblock: Jet Engine, Jet Filters" - yes, they have some shortcomings&simplifying when compared to Pods/FacetWP at some aspects, but its still the COMPLETE solution - custom content types with custom data tables with  filters for them with possibility to display everything at frontend - they whole data-chain works completely!

Thus your Pods now have not 1, but already 2 FUNDAMENTAL problems:
1) Issue #6000 - Pods fields settings storing at wp_posts, wp_postmeta - which wrong and creates garbage with too much SQL requests.
2) This issue - that NOT ONE filtering/search plugins does NOT SUPPORT Pods custom tables!!!
:(((

You really need to resolve 2 those problems asap, or for any huge/important custom databases Pods becomes just UNUSABLE! It's a specially pity and bad because exactly the Pods overall functionality is higher and with bigger potential! But who will use it without search/filtering possibility and with DB overload??? :(((

**Comment by justdudeone** [comment]
Guys, you also can add support for Pods Custom Tables for FacetWP by your own addon!
Without it Search&Filter its just impossible to use Pods Custom Tables! :((

**Comment by justdudeone** [comment]
Any progress with Filters&Search???

**Comment by sc0ttkclark** [comment]
Posting my replies here from elsewhere for posterity.

[filters()](https://docs.pods.io/code/pods/filters/) works with any pod type automatically. It searches any text based fields too. It's not a FacetWP replacement but it's a start. The main thing here is that while the UI is somewhat limited, the code underneath this is actually very powerful and you can build whatever UI you want to work with it yourself.

What you want is not possible with FacetWP because it only works with posts. There’s an add-on that works for users but it only works because it syncs/copies the data to a hidden user post type that makes it work. In the future, FacetWP may have its engine written to support other data sources but it is not flexible for that right now and that's OK, Matt didn't set out to make it something other than what it was for -- filtering posts/custom post types.

I know this is frustrating because you want that but it’s just not a possibility unless we build a FacetWP kind of feature into Pods or a Pods Pro by SKCDEV Add-On which would take significant time and investment to make happen.

FilterEverything seems like the only plugin that could be capable of integrating with Advanced Content Types in the future. We’d have to spend time writing a new WP_Query compatible API written so that it can easily translate to what their code needs.

There's no ETA on any of this. My hands are completely full with work on Pods 2.9 and then Pods 3.0. I am interested in writing code to make this work but it's lower on the priorities as set by our Friends of Pods who provide us the funding to do what we do.

Our Friends of Pods vote on what we work on for each major Pods release and it's what keeps things focused for us -- there will always be a ton of ideas from people on what they would like to see.
